File: backend/app/schemas/doctor.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime

class DoctorProfile(BaseModel):
    id: str
    name: str
    specialty: str
    rating: float
    experience: str
    avatar: str
    bio: str
    availability: List[str]
    price: str

    class Config:
        from_attributes = True

class ConsultationCreate(BaseModel):
    doctor_id: str
    date_time: str = Field(..., description="Selected appointment slot (e.g. Wednesday 2PM - 3PM)")

class ConsultationDB(BaseModel):
    id: str
    user_id: str
    doctor_id: str
    doctor_name: str
    doctor_specialty: str
    doctor_avatar: str
    date_time: str
    status: str = "Scheduled"
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/schemas/doctor.ts
export interface DoctorProfile {
  id: string;
  name: string;
  specialty: string;
  rating: number;
  experience: string;
  avatar: string;
  bio: string;
  availability: string[];
  price: string;
}

export interface ConsultationDB {
  id: string;
  user_id: string;
  doctor_id: string;
  doctor_name: string;
  doctor_specialty: string;
  doctor_avatar: string;
  date_time: string;
  status: string; // Scheduled, Completed, Cancelled
  created_at: string;
}
